lib/bumblebee/audio/speech_to_text_whisper.ex

```diff
@@ -76,101 +86,144 @@ defmodule Bumblebee.Audio.SpeechToTextWhisper do
+
+          true ->
+            stream =
+              Stream.map(other, fn
+                %Nx.Tensor{shape: {_}} = chunk ->
+                  Nx.backend_transfer(chunk, Nx.BinaryBackend)
+
+                item ->
+                  raise ArgumentError,
+                        "expected each enumerable item to be a 1-dimensional tensor, got: #{inspect(item)}"
+              end)
+
+            {:ok, stream}
+        end
     end
   end
 
-  defp maybe_stream(serving, false, spec, featurizer, tokenizer, timestamps?) do
-    Nx.Serving.client_postprocessing(serving, fn
-      {outputs, _metadata}, {multi?, all_num_chunks, lengths} ->
-        chunk_outputs = Nx.to_list(outputs)
-
-        all_num_chunks
-        |> Enum.map_reduce(chunk_outputs, fn num_chunks, chunk_outputs ->
-          {outputs, rest} = Enum.split(chunk_outputs, num_chunks)
-          state = decode_chunk_outputs_init(lengths, spec, featurizer, tokenizer)
-          {chunks, _state} = decode_chunk_outputs_update(state, outputs, timestamps?, tokenizer)
-          {%{chunks: chunks}, rest}
-        end)
-        |> elem(0)
-        |> Shared.normalize_output(multi?)
-    end)
-  end
+  defp ffmpeg_read_as_pcm(path, sampling_rate) do
+    channels = 1
 
-  defp maybe_stream(serving, true, spec, featurizer, tokenizer, timestamps?) do
-    serving
-    |> Nx.Serving.streaming()
-    |> Nx.Serving.client_postprocessing(fn stream, {false = _multi?, _all_num_chunks, lengths} ->
-      state = decode_chunk_outputs_init(lengths, spec, featurizer, tokenizer)
+    format =
+      case System.endianness() do
+        :little -> "f32le"
+        :big -> "f32be"
+      end
 
-      Stream.transform(stream, state, fn {:batch, outputs, _metadata}, state ->
-        outputs = Nx.to_list(outputs)
-        decode_chunk_outputs_update(state, outputs, timestamps?, tokenizer)
-      end)
-    end)
+    cond do
+      System.find_executable("ffmpeg") == nil ->
+        {:error, "ffmpeg not found in PATH"}
+
+      not File.exists?(path) ->
+        {:error, "no file found at #{path}"}
+
+      true ->
+        # This chunk can be of arbitrary size, the serving accumulates
+        # and overlaps chunks internally as needed. We read the file
+        # as stream to reduce memory usage
+        chunk_size = 30
+
+        stream =
+          Stream.iterate(0, fn offset -> offset + chunk_size end)
+          |> Stream.transform({}, fn offset, acc ->
+            System.cmd("ffmpeg", [
+              "-ss",
+              Integer.to_string(offset),
+              "-t",
+              Integer.to_string(chunk_size),
+              "-i",
+              path,
+              "-ac",
+              Integer.to_string(channels),
+              "-ar",
+              Integer.to_string(sampling_rate),
+              "-f",
+              format,
+              "-hide_banner",
+              "-loglevel",
+              "quiet",
+              "pipe:1"
+            ])
```

Comment:
```suggestion
            System.cmd(
              "ffmpeg",
              ~w[-ss #{offset} -t #{chunk_size} ... -hide_banner -loglevel quiet pipe:1]
            )
```

The above would also work if you prefer it.

Reply:
Pretty!